components/homme/src/share/prim_driver_base.F90

```diff
@@ -1331,8 +1344,15 @@ subroutine prim_step_flexible(hybrid, elem, nets, nete, dt, tl, hvcoord, compute
           end if
        end if
 
-       call prim_advance_exp(elem, deriv1, hvcoord, hybrid, dt, tl, nets, nete, &
-            compute_diagnostics_it)
+       if (.not. single_column) then
```

Comment:
my personal preference would be to have ifdefs here, since these loops will be evaluated every timestep (though maybe for negligible cost), but i think other people prefer if-statements to ifdefs.